crates/solver/src/settlement.rs

```diff
@@ -299,6 +303,78 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: u64,
+        tokens_to_satisfy_price_test: &Option<Vec<H160>>,
```

Comment:
Might be simpler for this function to take this as a HashSet. Driver would convert the command line argument into the hashset in it's `new` function.